src/sections/layer_and_mask_information_section/layer.rs

```diff
@@ -100,43 +159,37 @@ pub enum PsdLayerError {
 impl PsdLayer {
     /// Create a new photoshop layer
     pub fn new(
-        name: String,
-        layer_top: i32,
-        layer_left: i32,
-        layer_bottom: i32,
-        layer_right: i32,
+        layer_record: LayerRecord,
         psd_width: u32,
         psd_height: u32,
-        sub_layers: Option<PsdLayerGroup>,
+        group_id: Option<u32>,
     ) -> PsdLayer {
         PsdLayer {
-            name,
             channels: HashMap::new(),
-            layer_top,
-            layer_left,
-            layer_bottom,
-            layer_right,
-            psd_width,
-            psd_height,
-            sub_layers,
+            layer_properties: LayerProperties::from_layer_record(
+                layer_record,
+                psd_width,
+                psd_height,
+                group_id,
+            ),
         }
     }
 
     /// Get the name of the layer
     pub fn name(&self) -> &str {
-        &self.name
+        &self.layer_properties.name
     }
 
     /// The width of the layer
     pub fn width(&self) -> u16 {
         // If left is at 0 and right is at 4, the width is 5
-        (self.layer_right - self.layer_left) as u16 + 1
+        (self.layer_properties.layer_right - self.layer_properties.layer_left) as u16 + 1
```

Comment:
Same here - would remove the duplicated width/height methods and `impl Deref for PsdLayer` to point to a `LayerProperties`